rbx_binary/src/deserializer.rs

```diff
@@ -856,7 +856,45 @@ impl<R: Read> BinaryDeserializer<R> {
             Type::NumberRange => {}
             Type::Rect => {}
             Type::PhysicalProperties => {}
-            Type::Color3uint8 => {}
+            Type::Color3uint8 => match canonical_type {
+                VariantType::Color3 => {
+                    let len = type_info.referents.len();
+                    let mut r = Vec::with_capacity(len);
+                    let mut g = Vec::with_capacity(len);
+                    let mut b = Vec::with_capacity(len);
+
+                    for _ in 0..len {
+                        r.push(chunk.read_u8().unwrap());
+                    }
+
+                    for _ in 0..len {
+                        g.push(chunk.read_u8().unwrap());
+                    }
+
+                    for _ in 0..len {
+                        b.push(chunk.read_u8().unwrap());
+                    }
```

Comment:
Instead of writing these loops by hand, we can use Rust's [Read::read_exact](https://doc.rust-lang.org/stable/std/io/trait.Read.html#method.read_exact).

```rust
let mut r = vec![0; len];
chunk.read_exact(r.as_mut_slice())?;
```

...for each color.